### PR TITLE
16 migrate to angular mat

### PR DESCRIPTION
**Pull request by weihanli101**
Couple things to note:
- The showboard() function was not migrated over as I feel it will be replaced in the future anyways
- The default setting for mat-button-toggle-group apparently shows them all as selected until one is click (maybe something we can fix in the future? Or someone with more experience can)

### CODE REVIEW

File: src/styles.scss [hunk 1]
```diff
@@ -2,6 +2,7 @@
 //font imports for XEL
```
Comment (thechausenone) [inline src/styles.scss]: get rid of these fonts

File: src/index.html [hunk 2]
```diff
@@ -8,6 +8,7 @@
   <link rel="icon" type="image/x-icon" href="favicon.ico">
   <link rel="stylesheet" href="../node_modules/xel/stylesheets/material.theme.css">
```
Comment (thechausenone) [inline src/index.html]: remove lines 9 & 10

File: src/index.html [hunk 3]
```diff
@@ -8,6 +8,7 @@
...
   <script src="../node_modules/xel/xel.min.js"></script>
+  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
```
Comment (thechausenone) [inline src/index.html]: might want to consider downloading this and adding it to our project so we're not dependent on stuff from the web

File: package.json [hunk 4]
```diff
@@ -36,14 +36,17 @@
...
+    "@angular/core": "^4.4.6",
+    "@angular/forms": "^4.4.6",
+    "@angular/http": "^4.4.6",
+    "@angular/material": "^2.0.0-beta.12",
+    "@angular/platform-browser": "^4.4.6",
+    "@angular/platform-browser-dynamic": "^4.4.6",
+    "@angular/router": "^4.4.6",
     "angular-gridster2": "^3.15.1",
     "core-js": "2.4.1",
     "enhanced-resolve": "3.3.0",
```
Comment (thechausenone) [inline package.json]: remove the xel dependency from here (line 51/54)

**Comment by weihanli101** [comment]
Updated with changes didn't download the packages yet. Not sure how to update diff.

File: package.json [hunk 5]
```diff
@@ -36,24 +36,23 @@
...
+    "@angular/compiler": "^4.4.6",
+    "@angular/core": "^4.4.6",
+    "@angular/forms": "^4.4.6",
+    "@angular/http": "^4.4.6",
+    "@angular/material": "^2.0.0-beta.12",
+    "@angular/platform-browser": "^4.4.6",
+    "@angular/platform-browser-dynamic": "^4.4.6",
+    "@angular/router": "^4.4.6",
     "angular-gridster2": "^3.15.1",
-    "core-js": "2.4.1",
```
Comment (thechausenone) [inline package.json]: whyd u delete core-js, enhanced-resolve, and rxjs? revert pls

File: src/index.html [hunk 6]
```diff
@@ -5,9 +5,7 @@
   <title>AngularElectron</title>
   <base href="">
   <meta name="viewport" content="width=device-width, initial-scale=1">
-  <link rel="icon" type="image/x-icon" href="favicon.ico">
```
Comment (thechausenone) [inline src/index.html]: this was here since before xel was integrated, i think its the app's tab icon or something. i think we should keep it...

**Pull request merged**